go/tools/generate_test_main.go

```diff
@@ -82,33 +79,24 @@ func main() {
 		}
 	}
 
-{{ if .Version.go1_8 }}
+{{if eq .Version "go1.8"}}
```

Comment:
Won't this break when Go 1.9 is released? Should this be `if ge .Version "go1.8"` instead?